test/kad-dht.spec.js

```diff
@@ -6,707 +6,518 @@ chai.use(require('dirty-chai'))
-    ], (err) => {
-      expect(err).to.not.exist()
+  describe('start and stop', () => {
+    it('simple with defaults', async () => {
+      const dht = createDHT(peerInfos[0])
+
+      sinon.spy(dht.network, 'start')
+      sinon.spy(dht.randomWalk, 'start')
+
+      sinon.spy(dht.network, 'stop')
+      sinon.spy(dht.randomWalk, 'stop')
+
+      await dht.start()
+      expect(dht.network.start.calledOnce).to.equal(true)
+      expect(dht.randomWalk.start.calledOnce).to.equal(true)
+
+      await dht.stop()
       expect(dht.network.stop.calledOnce).to.equal(true)
       expect(dht.randomWalk.stop.calledOnce).to.equal(true)
-
-      done()
     })
-  })
 
-  it('should be able to start with random-walk disabled', function (done) {
-    const sw = new Switch(peerInfos[0], new PeerBook())
-    sw.transport.add('tcp', new TCP())
-    sw.connection.addStreamMuxer(Mplex)
-    sw.connection.reuse()
-    const dht = new KadDHT(sw, { randomWalk: { enabled: false } })
-
-    sinon.spy(dht.network, 'start')
-    sinon.spy(dht.randomWalk, 'start')
-
-    sinon.spy(dht.network, 'stop')
-    sinon.spy(dht.randomWalk, 'stop')
-
-    series([
-      (cb) => dht.start(cb),
-      (cb) => {
-        expect(dht.network.start.calledOnce).to.equal(true)
-        expect(dht.randomWalk._runningHandle).to.not.exist()
-
-        cb()
-      },
-      (cb) => dht.stop(cb)
-    ], (err) => {
-      expect(err).to.not.exist()
+    it('random-walk disabled', async () => {
+      const dht = createDHT(peerInfos[0], {
+        randomWalk: { enabled: false }
+      })
+
+      sinon.spy(dht.network, 'start')
+      sinon.spy(dht.randomWalk, 'start')
+
+      sinon.spy(dht.network, 'stop')
+      sinon.spy(dht.randomWalk, 'stop')
+
+      await dht.start()
+      expect(dht.network.start.calledOnce).to.equal(true)
+      expect(dht.randomWalk._runningHandle).to.not.exist()
+
+      await dht.stop()
       expect(dht.network.stop.calledOnce).to.equal(true)
       expect(dht.randomWalk.stop.calledOnce).to.equal(true) // Should be always disabled, as it can be started using the instance
-
-      done()
     })
-  })
 
-  it('should fail to start when already started', function (done) {
-    const sw = new Switch(peerInfos[0], new PeerBook())
-    sw.transport.add('tcp', new TCP())
-    sw.connection.addStreamMuxer(Mplex)
-    sw.connection.reuse()
-    const dht = new KadDHT(sw, {
-      randomWalk: {
-        enabled: false
+    // TODO: not fail!
+    it('fail when already started', async () => {
+      const dht = createDHT(peerInfos[0])
+
+      await dht.start()
+      try {
+        await dht.start()
+      } catch (err) {
+        expect(err).to.exist()
+        return
       }
+      throw new Error('should fail to start when already registered')
     })
 
-    series([
-      (cb) => dht.start(cb),
-      (cb) => dht.start(cb)
-    ], (err) => {
-      expect(err).to.exist()
-      done()
-    })
-  })
+    it('should fail to stop when was not started', () => {
```

Comment:
This should also not fail, since these are existing behaviors we can look at doing them in separate PRs since this is already massive.

Reply:
I will add it in the next PR, yes